src/main/java/com/develhope/spring/model/user/Rental.java

```diff
@@ -0,0 +1,47 @@
+package com.develhope.spring.model.user;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+public class Rental {
+    private LocalDate startDate;
+    private LocalDate endDate;
+    private BigDecimal dailyCost;
+    private BigDecimal totalCost;
+    private boolean paid;
+
+    //it was decided to use vehicle (object) but instead of using
+    // Vehicle as object is created private int vehicleId;
+    // passed to the constructor
+    // handled the same way in the Order class of User
+    private int vehicleId;
+
+
+    public Rental(LocalDate startDate, LocalDate endDate, BigDecimal dailyCost, int vehicleId) {
+        this.startDate = startDate;
+        this.endDate = endDate;
+        this.dailyCost = dailyCost;
+        this.vehicleId = vehicleId;
+        this.calculateTotalCost();
+        this.paid = false;
```

Comment:
Non metterei il pagamento di default a false. Il flag pagato idealmente va inserito sul momento della creazione dell'ordine, lo passerei al costruttore quindi. Così che nel momento che il noleggio viene fatto, se il pagamento è già stato fatto si mette true, se no false se deve ancora essere saldato

Reply:
hai il metodo sotto per cambiare lo stato.
Il ragionamento è partire da false e poterla settare in un secondo momento
hai più controllo

Reply:
Anche parlando con alessio concordiamo sul fatto che dato che l'ordine lo crei sul momento è più efficiente poter comunicare subito se magari è stato pagato, invece che segnarlo come tale in un secondo momento.